src/core/dfn.js

```diff
@@ -9,13 +9,9 @@ export const name = "core/dfn";
 
 export function run() {
   document.querySelectorAll("dfn").forEach(dfn => {
-    /** @type {HTMLElement} */
-    const closestDfn = dfn.closest("[data-dfn-for]");
-    if (closestDfn && closestDfn !== dfn && !dfn.dataset.dfnFor) {
-      dfn.dataset.dfnFor = closestDfn.dataset.dfnFor;
-    }
-    // TODO: we should probably use weakmaps and weaksets here to avoid leaks.
-    const titles = getDfnTitles(dfn, { isDefinition: true });
+    const titles = getDfnTitles(dfn);
     registerDefinition(dfn, titles);
+    if (titles.length) dfn.dataset.lt = titles.join("|");
```

Comment:
Found that this unexpectedly add `data-lt` everywhere, because `titles` nearly always contains an item here unless `data-lt-noDefault`.

Reply:
I think that's ok. The `lt`'s now include the normalized text content, which matches what BikeShed does, I think.

Reply:
Not sure I can find an example that bikeshed greedily adds `data-lt`, Inspecting Web IDL spec says it doesn't.

Reply:
I was looking at fetch, for instance:
```
<dfn class="dfn-paneled" data-dfn-type="dfn" data-export data-lt="collect an HTTP quoted string|collecting an HTTP quoted string" id="collect-an-http-quoted-string">collect an HTTP quoted string</dfn>
```

Reply:
That's 👍 and it needs `data-lt` to do that, but the ReSpec behavior here is:

```html
<dfn>hey</dfn>
<!-- becomes... -->
<dfn data-lt="hey">hey</dfn>
```

Which is weird.

Reply:
Ah, maybe it only does it for `data-export` things?

Reply:
The question is, is there harm in ReSpec always adding `data-lt`? As an editor, I only ever want to write the alternative linking names, but not the text content. So:

```
<dfn data-lt="yellow fruit">banana</dfn>
<!-- becomes... -->
<dfn data-lt="yellow fruit|banana">banana</dfn>
``` 

Would be ok, no? Or would that be harmful somehow?

Reply:
I think that's okay to do (regardless of what bikeshed do for that) and it's needed for Shepherd, but in the `hey` case Shepherd won't need it.

Reply:
Ok, I'll omit it to avoid the duplication.